spec/controllers/adopters_controller_spec.rb

```diff
@@ -318,7 +318,177 @@
+          expect(parse_response(response)['ci']).to eq(adopter.ci)
+          expect(parse_response(response)['first_name']).to eq(adopter.first_name)
+          expect(parse_response(response)['last_name']).to eq(adopter.last_name)
+          expect(parse_response(response)['email']).to eq(adopter.email)
+          expect(parse_response(response)['phone']).to eq(adopter.phone)
+          expect(parse_response(response)['house_description']).to eq(adopter.house_description)
+          expect(parse_response(response)['blacklisted']).to eq(adopter.blacklisted)
+          expect(parse_response(response)['home_address']).to eq(adopter.home_address)
+        end
+
+        it 'devuelve codigo 200' do
+          get :show, id: adopter.id, format: 'json'
+          expect(response.status).to eq(200)
+        end
+      end
+
+      context 'con un usuario sin permisos para editar adoptantes' do
+        before(:each) do
+          request.headers['X-USER-TOKEN'] = user2.authentication_token
+        end
+
+        it 'devuelve una array no vacio' do
+          get :show, id: adopter2.id, format: 'json'
+          expect(parse_response(response)).to_not be_nil
+        end
+
+        it 'devuelve una array con los datos del adoptante creado' do
+          get :show, id: adopter.id, format: 'json'
+          expect(parse_response(response)['id']).to eq(adopter.id)
+          expect(parse_response(response)['ci']).to eq(adopter.ci)
+          expect(parse_response(response)['first_name']).to eq(adopter.first_name)
+          expect(parse_response(response)['last_name']).to eq(adopter.last_name)
+          expect(parse_response(response)['email']).to eq(adopter.email)
+          expect(parse_response(response)['phone']).to eq(adopter.phone)
+          expect(parse_response(response)['house_description']).to eq(adopter.house_description)
+          expect(parse_response(response)['blacklisted']).to eq(adopter.blacklisted)
+          expect(parse_response(response)['home_address']).to eq(adopter.home_address)
+        end
+
+        it 'devuelve codigo 200' do
+          get :show, id: adopter2.id, format: 'json'
+          expect(response.status).to eq(200)
+        end
+      end
+    end
+  end
+
+  describe 'GET #search' do
+    context 'se listan los adoptantes buscados' do
+
+      let!(:adToSearch) { create(:adopter, first_name: 'Miguel', last_name: 'Viera') }
+      let!(:adToSearch2) { create(:adopter, first_name: 'Charly', last_name: 'Sosa', blacklisted: 'true') }
+      let!(:adToSearch3) { create(:adopter, first_name: 'Char', last_name: 'Mate', blacklisted: 'true') }
+      let!(:adToSearch4) { create(:adopter, first_name: 'Charly', last_name: 'Sosa', blacklisted: 'false') }
+
+      before(:each) do
+        request.headers['X-USER-TOKEN'] = user.authentication_token
+      end
+
+      context 'por nombre especifico' do
+        it 'devuelve codigo 200' do
+          get :search, name: 'Miguel', format: 'json'
```

Comment:
A esta linea de codigo que aparece en los siguientes 3 it `get :search, name: 'Miguel', format: 'json'` no se podría poner en un bloque each para el context 'por nombre especifico'?. Este comentario aplica para el resto del Archivo. Todo lo que se pueda reutilizar para no repetir codigo mejor.